runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java

```diff
@@ -35,23 +35,33 @@ final public class StreamAddressSpace {
     // Holds the complete map of addresses for this stream.
     private Roaring64NavigableMap bitmap;
 
+    // True if caching should be used in internal StreamAddressSpace data structures.
+    // Note: this parameter is not serialized, nor used to determine equality.
+    private final boolean enableCaching;
+
     /**
      * This constructor is required to facilitate deserialization, keep it private.
      * The internal bitmap container shouldn't be exposed to external consumers.
      * @param trimMark Stream's trim mark
      * @param bitmap Stream's bitmap
+     * @param enableCaching True if and only if caching is enabled in the stream's bitmap
      */
-    private StreamAddressSpace(long trimMark, Roaring64NavigableMap bitmap) {
+    private StreamAddressSpace(long trimMark, Roaring64NavigableMap bitmap, boolean enableCaching) {
         this.trimMark = trimMark;
         this.bitmap = bitmap;
+        this.enableCaching = enableCaching;
     }
 
     public StreamAddressSpace() {
-        this(Address.NON_ADDRESS, Roaring64NavigableMap.bitmapOf());
+        this(Address.NON_ADDRESS, Roaring64NavigableMap.bitmapOf(), true);
+    }
+
+    public StreamAddressSpace(boolean enableCaching) {
+        this(Address.NON_ADDRESS, new Roaring64NavigableMap(false, enableCaching), enableCaching);
     }
 
     public StreamAddressSpace(long trimMark, Set<Long> addresses) {
-        this(trimMark, Roaring64NavigableMap.bitmapOf(Longs.toArray(addresses)));
+        this(trimMark, Roaring64NavigableMap.bitmapOf(Longs.toArray(addresses)), true);
```

Comment:
I wonder if its safer to invert the default behavior so that the default constructor is thread-safe for read only access, I'm just worried that future uses of this Type will introduce subtle bugs.

Reply:
Good idea - I have updated the default constructor to disable internal caching, along with relevant invocations.